apps/web/app/(app)/environments/[environmentId]/surveys/[surveyId]/(analysis)/summary/LinkSingleUseSurveyModal.tsx

```diff
@@ -0,0 +1,122 @@
+"use client";
+
+import { Dialog, DialogContent } from "@formbricks/ui";
+import { TSurvey } from "@formbricks/types/v1/surveys";
+import { Button } from "@formbricks/ui";
+import { ArrowPathIcon, CheckIcon } from "@heroicons/react/24/outline";
+import { CheckCircleIcon, DocumentDuplicateIcon, EyeIcon } from "@heroicons/react/24/solid";
+import { useRef, useState } from "react";
+import toast from "react-hot-toast";
+import cuid2 from "@paralleldrive/cuid2";
+import { truncateMiddle } from "@/lib/utils";
+import { cn } from "@formbricks/lib/cn";
+
+interface LinkSingleUseSurveyModalProps {
+  survey: TSurvey;
+  open: boolean;
+  setOpen: (open: boolean) => void;
+}
+
+const generateSingleUseIds = () =>
+  Array(5)
+    .fill(null)
+    .map(() => cuid2.createId());
+
+export default function LinkSingleUseSurveyModal({ survey, open, setOpen }: LinkSingleUseSurveyModalProps) {
+  const defaultSurveyUrl = `${window.location.protocol}//${window.location.host}/s/${survey.id}`;
+  const initialSingleUseIds = generateSingleUseIds();
+  const [selectedSingleUseIds, setSelectedSingleIds] = useState<number[]>([]);
+
+  const linkTextRef = useRef<HTMLDivElement>(null);
+  const [singleUseIds, setSingleUseIds] = useState(initialSingleUseIds);
+
+  const handleLinkOnClick = (index: number) => {
+    setSelectedSingleIds([...selectedSingleUseIds, index]);
+    const surveyUrl = `${defaultSurveyUrl}?suId=${singleUseIds[index]}`;
+    navigator.clipboard.writeText(surveyUrl);
+    toast.success("URL copied to clipboard!");
+  };
+
+  return (
+    <Dialog open={open} onOpenChange={setOpen}>
+      <DialogContent className="bottom-0 max-w-sm bg-white p-4 sm:bottom-auto sm:max-w-xl sm:p-6">
+        <div className="mx-auto flex h-12 w-12 items-center justify-center rounded-full bg-teal-100">
+          <CheckIcon className="h-6 w-6 text-teal-600" aria-hidden="true" />
+        </div>
+        <div className="mt-3 text-center sm:mt-5">
+          <h3 className="text-lg font-semibold leading-6 text-gray-900">Your survey is ready!</h3>
+          <div className="mt-4">
+            <p className="text-sm text-gray-500">
+              Here are 5 single use links to let people answer your survey:
+            </p>
+            <div ref={linkTextRef}>
+              {singleUseIds.map((singleUseId, index) => {
+                const isSelected = selectedSingleUseIds.includes(index);
+                return (
+                  <div
```

Comment:
We can add `key={singleUseId}` to avoid `Warning: Each child in a list should have a unique "key" prop`